src/overlays/actors/ovl_En_Bee/z_en_bee.c

```diff
@@ -118,121 +118,124 @@ void EnBee_Destroy(Actor* thisx, PlayState* play) {
 
-    Math_Vec3f_Copy(&this->unk_21C[0], &sp48);
-    Math_Vec3f_Copy(&sp48, &this->actor.home.pos);
+    Math_Vec3f_Copy(&this->moveToDestinations[0], &tmpPos);
+    Math_Vec3f_Copy(&tmpPos, &this->actor.home.pos);
 
-    sp48.x += Math_SinS(sp46 - 0x4000) * 50.0f;
-    sp48.y = Rand_ZeroFloat(50.0f) + (this->actor.floorHeight + 30.0f);
-    sp48.z += Math_CosS(sp46 - 0x4000) * 50.0f;
+    tmpPos.x += Math_SinS(yawOffset - 0x4000) * 50.0f;
+    tmpPos.y = Rand_ZeroFloat(50.0f) + (this->actor.floorHeight + 30.0f);
+    tmpPos.z += Math_CosS(yawOffset - 0x4000) * 50.0f;
 
-    Math_Vec3f_Copy(&this->unk_21C[1], &sp48);
+    Math_Vec3f_Copy(&this->moveToDestinations[1], &tmpPos);
 
-    this->unk_204 = Rand_S16Offset(0x14, 0x1E);
-    this->unk_208 = 0;
-    this->actionFunc = func_80B5A9E8;
+    this->attackDelayTimer = Rand_S16Offset(0x14, 0x1E);
+    this->isHostile = 0;
+    this->actionFunc = EnBee_Idle;
 }
 
-void func_80B5A9E8(EnBee* this, PlayState* play) {
-    Vec3f sp3C;
+void EnBee_Idle(EnBee* this, PlayState* play) {
+    Vec3f nextDestination;
     s32 pad[2];
 
     if ((this->actor.category != ACTORCAT_ENEMY) && (ActorCutscene_GetCurrentIndex() == -1)) {
         func_800BC154(play, &play->actorCtx, &this->actor, ACTORCAT_ENEMY);
     }
 
-    Math_Vec3f_Copy(&sp3C, &this->unk_21C[this->unk_214]);
-    sp3C.x += Math_SinS(this->unk_20C) * 30.0f;
-    sp3C.z += Math_CosS(this->unk_20C) * 30.0f;
+    Math_Vec3f_Copy(&nextDestination, &this->moveToDestinations[this->destinationIndex]);
+    nextDestination.x += Math_SinS(this->targetYaw) * 30.0f;
+    nextDestination.z += Math_CosS(this->targetYaw) * 30.0f;
 
-    if (!(this->unk_218 & 1)) {
-        this->unk_20C += (s16)((s32)randPlusMinusPoint5Scaled(1000.0f) + 4000);
+    if (!(this->instanceId & 1)) {
+        this->targetYaw += (s16)((s32)randPlusMinusPoint5Scaled(1000.0f) + 4000);
     } else {
-        this->unk_20C -= (s16)((s32)randPlusMinusPoint5Scaled(1000.0f) + 4000);
+        this->targetYaw -= (s16)((s32)randPlusMinusPoint5Scaled(1000.0f) + 4000);
     }
 
-    this->unk_210 += 1000;
-    this->actor.velocity.y = Math_SinS(this->unk_210);
+    this->flightHoverOffset += 1000;
+    this->actor.velocity.y = Math_SinS(this->flightHoverOffset);
 
-    if (this->unk_20C > 0x10000) {
-        this->unk_20C = 0;
-        this->unk_214++;
-        this->unk_214 &= 1;
+    if (this->targetYaw > 0x10000) {
+        this->targetYaw = 0;
+        this->destinationIndex++;
+        this->destinationIndex &= 1;
     }
 
-    Math_SmoothStepToS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &sp3C), 1, 0x7D0, 0);
+    Math_SmoothStepToS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &nextDestination), 1, 0x7D0, 0);
     Math_ApproachF(&this->actor.speedXZ, 3.0f, 0.3f, 1.0f);
 
-    if ((this->unk_204 == 0) && (this->actor.params != 0)) {
-        func_80B5ABC4(this);
+    if ((this->attackDelayTimer == 0) && (this->actor.params != 0)) {
```

Comment:
From what I could see in KZ, it looks like bees with `params = 0` the bee will be in a `idle/docile` state. If you set `params = 1` the bee will turn to be aggressive and pursue Link. You could probably add an enum in the header to cover these two states. You can look at `z_en_crow` or `z_en_firefly` for an example.

Reply:
I'm not really sure it makes sense to do this, the actor itself only ever has the one check against its params - to be nonzero - and the only other place bees are spawned (the beehive actor) it sets the param as either 1 or 2.. so I don't think there's enough information to give names to the states. and having an enum for 0 seems kinda silly.

so going to resolve this, please re-open if you disagree with my reasoning here